docs/recipes/recipe-data-structures-in-git/index.md

```diff
@@ -0,0 +1,366 @@
+3:00PM WARN validation file=data-structures/com.example/login.yaml
+  messages=
+  │ The schema is missing the "description" property (/properties/result)
+  │ The schema is missing the "description" property (/)
+```
+
+### Publish to development
+
+Apart from the missing descriptions everything looks good. We can fill them in later™. Let's go ahead and publish our data structure to our [development](/docs/managing-data-quality/testing-and-qa-workflows/) environment.
+
+```bash
+$ snowplow-cli ds publish dev
+```
+:::tip
+We omit the directory here but as with other commands the default directory will get used and it will attempt to publish any data structures it can find.
+:::
+
+
+The command should output something close to the following:
+
+```
+3:00PM INFO publishing to dev from paths=[data-structures]
+3:00PM INFO will create file=data-structures/com.example/login.yaml vendor=com.example name=login version=1-0-0
+3:00PM WARN validation file=data-structures/com.example/login.yaml
+  messages=
+  │ The schema is missing the "description" property (/properties/result)
+  │ The schema is missing the "description" property (/)
+3:00PM INFO all done!
+```
+
+:::note
+Publishing to `dev` will also run validation. It will only fail on ERROR notifications.
+:::
+
+You should now be able to see your published data structure in [BDP Console UI](https://console.snowplowanalytics.com/data-structures). If you click through from the data structure listing to view the `login` data structure you should see the following banner.
+
+![](./images/locked.png)
+
+Any data structures published using snowplow-cli will automatically get this banner and have UI based editing disabled. It is a good idea to settle on one source of truth for each data structure to avoid potential conflicts.
+
+### Publish to production
+
+With our data structure deployed to develop and working as we expect we can safely publish it to production.
+
+```bash
+$ snowplow-cli ds publish prod
+```
+```
+3:00PM INFO publishing to prod from paths=[data-structures]
+3:00PM INFO will update file=data-structures/com.example/login.yaml local=1-0-0 remote=""
+3:00PM INFO all done!
+```
+:::note
+Data structures must be published to `dev` before they can be published to `prod`
+:::
+
+We have now seen how to create, validate and then publish a new data structure from the command line. Next we'll look at how to configure github actions to run validation and publishing automatically for us.
+
+
+## Automating with github actions
+
+### Set up repository
+
+We'll not go into the details of creating github repositories and initial commits here. The [github docs](https://docs.github.com/) do an excellent job of that already. The next few steps will assume a working github repository containing the directory and data structure we created in the previous section. It will have two branches named `main` and `develop` which should be in sync.
```

Comment:
```suggestion
We'll not go into the details of creating github repositories and initial commits here, the [github docs](https://docs.github.com/) provide full guidance how to do those. The next few steps will assume a working github repository containing the directory and data structure we created in the previous section. It will have two branches named `main` and `develop` which should be in sync.
```

Reply:
At primary school I got roasted for writing extremely long sentences and using commas when I should have used full stops. That roasting has lived with me ever since and now every time I feel like I want to comma I find a way to full stop.